python/cuda_parallel/cuda/parallel/experimental/iterators/_iterators.py

```diff
@@ -372,3 +417,20 @@ def dereference(state):
             return it_dereference(state)
 
     return AdvancedIterator(it, offset)
+
+
+def get_last_element_ptr(device_array) -> int:
```

Comment:
Just a nit: if we don't intend for this function to be used outside this module, lets add a `_` prefix to the name.